### PR TITLE
[WFLY-2217] - enable reload tests, disabled due to reloading op

### PR DESCRIPTION
**Pull request by baranowb**
https://issues.jboss.org/browse/WFLY-2217

some tests are still disabled because of "reloading issues". This is no longer true. This PR enables two. 
https://issues.jboss.org/browse/WFLY-5606 handle another one.

Upstream: https://issues.jboss.org/browse/WFLY-2217 , https://issues.jboss.org/browse/WFLY-5606
EAP7.x: https://issues.jboss.org/browse/JBEAP-2160
EAP6.4.x: https://bugzilla.redhat.com/show_bug.cgi?id=900065

PRs:
Upstream: wildfly/wildfly#8437 , wildfly/wildfly#8367
EAP7.x: https://github.com/jbossas/jboss-eap7/pull/738
EAP 6.4.x: none atm.

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java [hunk 1]
```diff
@@ -87,6 +89,30 @@
...
+            // check that the connector is down
+            Assert.assertFalse("Could not connect to created connector.",WebUtil.testHttpURL(new URL(
+                    "http", testHost, TEST_PORT, "/").toString()));
+        } finally {
+            // remove socket binding
+            op = createOpNode("socket-binding-group=standard-sockets/socket-binding=test-binding", REMOVE);
+            result = executeOperation(op);
+    
+            // remove interface
+            op = createOpNode("interface=test-interface", REMOVE);
```
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java]: all this ops should be called with allow-resource-service-restart header so they don't need reload call at all and are removed at runtime.
see https://github.com/wildfly/wildfly/pull/8422/files#diff-e295371e5f22f6bc467299bf912c1faaR85
for example

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java [hunk 2]
```diff
@@ -129,9 +154,7 @@ public void testAddUpdateRemove() throws Exception {
 
         logger.info("Restarting server.");
 
-        // reload server
-        op = createOpNode(null, "reload");
-        result = executeOperation(op);
+        ServerReload.executeReloadAndWaitForCompletion(getModelControllerClient());
```
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java]: all resources you are testing are runtime addable/removable so reload is not needed at all.

Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java]: The socket-binding will be bound at this point, so this reload will be necessary. If it wasn't the assert at L128 of the current code in master would fail.

**Comment by bstansberry** [comment]
@maeste Please approve the JCA changes. Thanks!

@baranowb This will need a JBEAP to get merged any time soon, unless the BZ gets acked. Either way a backport to jboss-eap7 will be needed.

**Comment by maeste** [comment]
I'm good w/ those changes

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java (hunk 2 above)
Comment (ctomc) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java]: @bstansberry yes but if you call add for socket-binding with allow-resource-service-restart header it will be added on execution of that operation.
We do have few tests in our testsuite that verify this, for example ListenerTestCase adds/removes (new) socket bindings & listeners all without need for reload.

**Comment by baranowb** [comment]
This is not my test, I just fixed failure after it has been enabled in EAP/WFLY. If filed relevant PRs to 6 and 7

**Comment by baranowb** [comment]
ach, nope, scratch that. Different BZ/JIRA. Will file required PRs/JIRAs.

File: connector/src/main/java/org/jboss/as/connector/subsystems/jca/BootstrapContextAdd.java [hunk 3]
```diff
@@ -59,13 +59,16 @@ protected void performRuntime(final OperationContext context, final ModelNode op
         String name = JcaBootstrapContextDefinition.BootstrapCtxParameters.NAME.getAttribute().resolveModelAttribute(context, model).asString();
         String workmanager = JcaBootstrapContextDefinition.BootstrapCtxParameters.WORKMANAGER.getAttribute().resolveModelAttribute(context, model).asString();
         boolean usingDefaultWm = false;
+        CloneableBootstrapContext ctx;
         if (DEFAULT_NAME.equals( workmanager)) {
             usingDefaultWm = true;
+            ctx = new NamedBootstrapContext(name);
+        } else {
+            ctx = new NamedBootstrapContext(name,workmanager);
```
Comment (jamezp) [inline connector/src/main/java/org/jboss/as/connector/subsystems/jca/BootstrapContextAdd.java]: Why are these changes required? The referenced JIRA's only mention issues with tests yet both this and the change above don't seem to be in the tests.

**Comment by kabir** [comment]
@baranowb the 7.x PR link seems wrong

File: connector/src/main/java/org/jboss/as/connector/subsystems/jca/BootstrapContextAdd.java (hunk 3 above)
Comment (baranowb) [inline connector/src/main/java/org/jboss/as/connector/subsystems/jca/BootstrapContextAdd.java]: Because without those changes reload does not work for affected resources.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java (hunk 1 above)
Comment (baranowb) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/api/SocketsAndInterfacesTestCase.java]: Does not work in this case.

**Comment by baranowb** [comment]
hmm, I did not close it

**Comment by baranowb** [comment]
Seems like it is stuck as is, new PR.
